### PR TITLE
Vidhance context is now freed in destructor.

### PR DESCRIPTION
**Pull request by emilwestergren**
@sebastianbaginski peer review

### CODE REVIEW

File: HAL/CameraWrapper.cpp [hunk 1]
```diff
@@ -21,24 +21,25 @@ void _recycleGraphicBuffer(void* buffer) {
...
-		vidhance_motion_sensor_settings_set_rotation(sensorSettings, rotationSensor);
+	// Base settings object. Ownership is passed to Vidhance when context is created, hence memory management is handled internally.
+	vidhance_settings_t settings = vidhance_settings_new();
+	vidhance_motion_settings_t motionSettings = vidhance_settings_get_motion(settings);
+	vidhance_motion_sensor_settings_t sensorSettings = vidhance_motion_settings_get_sensor(motionSettings);
+	vidhance_motion_sensor_t gyroSensor = vidhance_gyro_sensor_new(GyroReader::getAngularVelocity, GyroReader::getInstance()->getSamplePeriod());
+	vidhance_motion_sensor_t rotationSensor = vidhance_rotation_sensor_new(gyroSensor);
+	vidhance_motion_sensor_settings_set_rotation(sensorSettings, rotationSensor);
 
-		context = vidhance_context_new(settings);
```
Comment (erikhagglund) [inline HAL/CameraWrapper.cpp]: remove the global variable?

Comment (emilwestergren) [inline HAL/CameraWrapper.cpp]: true dat

**Comment by emilwestergren** [comment]
Updated

**Comment by emilwestergren** [comment]
@erikhagglund peer review?

**Comment by erikhagglund** [comment]
now I am happy :)

**Pull request merged**